src/main/java/org/yagorundy/gui/components/SortingAlgorithmSelector.java

```diff
@@ -0,0 +1,41 @@
+package org.yagorundy.gui.components;
+
+import javax.swing.ButtonGroup;
+import javax.swing.JPanel;
+import javax.swing.JRadioButton;
+
+import org.yagorundy.gui.constants.GuiConstants;
+import org.yagorundy.gui.reusable.RadioButton;
+
+public class SortingAlgorithmSelector extends JPanel {
+    /**
+     *
+     */
+    private static final long serialVersionUID = -456708946790248002L;
+
+    private RadioButton[] buttons;
+    private ButtonGroup buttonGroup;
+
+    public SortingAlgorithmSelector(String[] algorithms) {
```

Comment:
Why don't you create algorithms enum, since you have fixed algorithm amout

Reply:
I used reflection to avoid typization (e.g. enums) 😄
I have a fixed ammount of algorithms, but if I want to add more, I can just create some classes without needing to touch anything in the code 😃